bundles/core/src/main/java/com/adobe/cq/wcm/core/components/services/link/LinkBuilder.java

```diff
@@ -0,0 +1,94 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2022 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.services.link;
+
+import java.util.Map;
+
+import org.apache.sling.api.resource.Resource;
+
+import com.adobe.cq.wcm.core.components.models.Page;
+
+/**
+ * Interface used to hold the configuration of a link.
+ */
+public interface LinkBuilder {
+
+    /**
+     * Returns the resource holding the link properties.
+     */
+    Resource getResource();
+
+    /**
+     * Sets the resource holding the link properties.
+     *
+     * @param resource The resource holding the link properties.
+     * @return the link builder holding the link configuration.
+     */
+    LinkBuilder setResource(Resource resource);
```

Comment:
Page, Resource and references which are used to create the Link should be passed as arguments to different constructors as they are exclusive. It makes no sense for example to build the Link with setResource and setTarget on one LinkBuilder instance.